src/sqlfluff/core/rules/analysis/select.py

```diff
@@ -64,7 +64,7 @@ def get_select_statement_info(
                     seen_using = True
                 elif seg.is_type("join_on_condition"):
                     for on_seg in seg.segments:
-                        if on_seg.is_type("expression"):
+                        if on_seg.is_type("bracketed", "expression"):
```

Comment:
`join_on_condition` contains `OptionallyBracketed(Ref("ExpressionSegment"))`, hence this code needs to handle brackets.